gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/EnvironmentAnalyticsResource.java

```diff
@@ -232,31 +231,36 @@ public String field() {
      */
     private String buildFieldFilterForNonAdmin(ExecutionContext executionContext, AnalyticsParam analyticsParam)
         throws FieldFilterEmptyException {
+        String fieldName;
+        Set<String> ids;
         // add filter by Apis or Applications
         if (isAdmin()) {
-            return null;
-        }
-
-        String fieldName;
-        List<String> ids;
-        if (APPLICATION_FIELD.equalsIgnoreCase(analyticsParam.getField())) {
-            fieldName = APPLICATION_FIELD;
-            ids =
-                applicationService
-                    .findIdsByUser(executionContext, getAuthenticatedUser())
-                    .stream()
-                    .filter(appId -> permissionService.hasPermission(executionContext, APPLICATION_ANALYTICS, appId, READ))
-                    .collect(Collectors.toList());
+            if (APPLICATION_FIELD.equalsIgnoreCase(analyticsParam.getField())) {
+                fieldName = APPLICATION_FIELD;
+                ids = applicationService.findIdsByOrganization(executionContext.getOrganizationId());
```

Comment:
Shouldn't we look for applications in the current environment instead?

Reply:
Didn't think about it. Is an application scoped by env ?

Reply:
yes

Reply:
I have updated the code to use searchIds methods and use the same ApplicationQuery than findIdsByUser.